AG-11297 Navigator DOM fallback slider

olegat opened the pull request:
https://ag-grid.atlassian.net/browse/AG-11297

olegat committed:
AG-11297 Add slider proxy inputs

olegat committed:
AG-11297 Rename textContext to textContent

olegat committed:
AG-11297 Rewrite proxyInteractionService.ts to avoid 'as'

olegat committed:
AG-11297 Add proxy sliders in navigator

olegat committed:
AG-11297 Use divs with aria roles in the navigator

olegat committed:
AG-11297 Use navigator's visible mask range for the proxy element

olegat committed:
AG-11297 Remove keyboard handling logic in navigator

This is now handled by the browser / OS.

olegat committed:
AG-11297 Remove 'tab-start' event

This no longer used (it was used by the legend and navigator).

olegat committed:
AG-11297 Remove HTMLElements when destroying navigator

olegat committed:
AG-11297 Make maskVisibleRange readonly

olegat committed:
AG-11297 Remove RegionManager.addRegionFromProperties

It's no longer necessary

olegat committed:
AG-11297 Make the navigator min/max handles actually sliders

olegat committed:
AG-11297 Sync proxy slider values and navigator min/max handles

olegat committed:
AG-11297 Update focus indicator when sliders move

olegat committed:
AG-11297 Use a proxy slider for panning too

olegat committed:
AG-11297 Cleanup Navigator.onSliderChange

alantreadway commented on packages/ag-charts-community/src/util/number.ts:
```diff
@@ -60,3 +60,20 @@ export function countFractionDigits(value: number, maximumFractionDigits = 10) {
+ * @param locale The locale to use to format the number.
+ */
+export function formatNormalizedPercentage(ratio: number, locale?: string) {
+    locale = locale || navigator.language;
```
Nit: Prefer `??` here?

olegat replied:
done.

alantreadway reviewed (commented):
LGTM, just one minor nit

olegat committed:
AG-11297 Cleanup: make proxyInteractionService.ts more readable

olegat committed:
AG-11297 CR: use ?? instead of ||

olegat committed:
AG-11297 Use <T> in createProxyElement so that typescript can infer the return type

olegat committed:
AG-11297 Fix build

olegat committed:
AG-11297 Make navigator a role="scrollbar" instead role="toolbar"

This seems to work better with screenreaders (tested in VoiceOver). The
screenreader expected the scrollbar to remain stationary, but have child
sliders that move around.

olegat committed:
AG-11297 nx format

alantreadway reviewed (approved):
LGTM
